terraform-providers: various

zowoq opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

zowoq committed:
terraform-providers: remove explicit version

not needed when we can just remove v from the front of rev

zowoq committed:
terraform-providers: add homepage that can be used as provider-source-address

also proper urls in the json

zowoq committed:
terraform-providers: simplify update script

zowoq committed:
terraform-providers: add license/spdx

zowoq committed:
terraform-providers: remove providers with unclear licensing

https://api.github.com/repos/logicmonitor/terraform-provider-logicmonitor/license
https://api.github.com/repos/vmware/terraform-provider-nsxt/license
https://api.github.com/repos/ondrejsika/terraform-provider-vercel/license

zowoq commented:
Manually managing licences is a maintenance burden when it can be resolved by a simple upstream fix. Removing packages without a working `spdx_id` for now, they can be added back when they have been fixed upstream.

`gitlab` is reading the license from the "old" github repo, it's correct at the moment and hopefully it won't change. Adding proper support for providers hosted on gitlab isn't worth the hassle for one provider.

Mic92 commented on pkgs/applications/networking/cluster/terraform-providers/default.nix:
```diff
@@ -87,17 +87,24 @@ let
     in
     lib.optionalAttrs config.allowAliases {
       b2 = removed "b2" "2022/06";
+      checkly = license "checkly" "2022/11";
```
I wonder what it need for github to pickup the license correctly: https://github.com/checkly/terraform-provider-checkly/blob/main/LICENSE

Mic92 commented on pkgs/applications/networking/cluster/terraform-providers/default.nix:
```diff
@@ -87,17 +87,24 @@ let
+      checkly = license "checkly" "2022/11";
       checkpoint = removed "checkpoint" "2022/11";
+      cloudscale = license "cloudscale" "2022/11";
+      dhall = license "dhall" "2022/11";
```
cc @baloo

baloo replied:
GitHub seems to be pedantic about the license formatting. Not sure that's the 80 columns or what.
https://github.com/awakesecurity/terraform-provider-dhall/commit/a00787e90547c6bf2e7e6ad5571d7b304a85aa1e
That fixed it.
```
$ curl -L -s "https://api.github.com/repos/awakesecurity/terraform-provider-dhall/license" | jq -r '.license.spdx_id'
BSD-3-Clause
```

I released a new 0.0.5 which should be up on hashicorp registry in a few minutes. @zowoq if you could re-run a `pkgs/applications/networking/cluster/terraform-providers/update-provider awakesecurity/dhall` that'd be appreciated.


@Mic92 thanks for the notification!

Mic92 commented on pkgs/applications/networking/cluster/terraform-providers/providers.json:
```diff
@@ -804,15 +768,6 @@
     "vendorHash": "sha256-6ePPxdULuTzLdVzzr12BjLu/lBN+5yIUq8U8FVUw/PM="
   },
-  "nsxt": {
-    "hash": "sha256-TOoRtCKdR1fBjk39dbMgBd7pDJGfjvkQAqfpJzWRwRg=",
```
This one also had a license but probably not picked up because it came without a .md extension: https://github.com/vmware/terraform-provider-nsxt/blob/master/LICENSE.txt

Mic92 commented on pkgs/applications/networking/cluster/terraform-providers/providers.json:
```diff
@@ -1221,15 +1176,6 @@
-    "hash": "sha256-/LHyNxal5Il/UzXdCKfVRzK/VVfSYMgoeKerWsedmho=",
-    "homepage": "https://registry.terraform.io/providers/ondrejsika/vercel",
-    "owner": "ondrejsika",
-    "repo": "terraform-provider-vercel",
```
This case is even weirder. Even github detect the license here: https://github.com/vercel/terraform-provider-vercel/blob/main/LICENSE

zowoq replied:
https://api.github.com/repos/ondrejsika/terraform-provider-vercel/license

Mic92 commented:
I pinged the few terraform provider that had no license at all.

zimbatm commented on pkgs/applications/networking/cluster/terraform-providers/default.nix:
```diff
@@ -17,15 +17,18 @@ let
      , rev
-     , version
+     , spdx ? "UNSET"
+     , version ? "${lib.removePrefix "v" rev}"
```
```suggestion
     , version ? lib.removePrefix "v" rev
```
the interpolation is not needed here

Mic92 commented:
Cloudscale's terraform provider was also fixed. Not released though: https://github.com/cloudscale-ch/terraform-provider-cloudscale/commit/6b8a3616ac6da1b364562584c90e7e9709e71da3

zowoq commented:
https://github.com/logicmonitor/terraform-provider-logicmonitor
https://github.com/ondrejsika/terraform-provider-vercel
These two don't have a license but it's a simple fix.

https://github.com/vmware/terraform-provider-nsxt
License but no `spdx_id`, the other vmware terraform repos do have a `spdx_id` so upstream should be fine with fixing it.

https://github.com/checkly/terraform-provider-checkly
Actually seems like the github api is broken for this one, it has a license but it's ignored? The repo is fork but it's doesn't actually point to the original repo. It was also added to nixpkgs this year so it's likely still being used, I've added an override for it as I doubt it can be fixed in the repo. Guess I'll open a ticket with github.

I'm planning to merge this in ~24hrs.
